lib/libvmod_vtc/vmod_vtc.vcc

```diff
@@ -169,6 +169,20 @@ version ``v1``).
 Candidate for moving into vmod_proxy, but there were concerns about
 the interface design
 
+$Function DURATION benchmark(SUB, INT total=1000, INT leg=10)
+
+Benchmark a VCL sub and return the average runtime.
+
+Initially and after every *leg* invocations, the task is rolled back
+in order to avoid workspace overflows, unless the workspace is already
+overflowed, in which case the benchmark is aborted. This implies that
+**the sub to be benchmarked can not use any VCL-generated data or
+PRIV_TASK state**.
+
+The run time is averaged over *total* runs. The actual total may be up
+to *leg* - 1 more, in other words, there are always *leg* calls
+executed before *total* is checked.
+
```

Comment:
Pretty sure I wouldn't have understood this without reading the code. The point here is that rollbacks are probably necessary if you repeatedly call a VCL sub, which I probably wouldn't have thought about at first, but it's easy to see once it's pointed out. I suggest using the "inner" vs "outer" terminology to explain the difference -- inner iterations without rollbacks, and a rollback for each outer iteration.

The second paragraph threw me off, first because I initially thought it would be *leg* * *total* results (maybe say something like "each inner iteration counts towards the *total*"). And it seems strange that could ever be less than *total* runs, until you realize that the benchmark might have been aborted due to WS overflow, except you currently can't tell whether that happened (see the comment above).

Reply:
addressed